kafka.tf

```diff
@@ -42,38 +42,22 @@ data "template_file" "cp-helm-charts-values" {
     domain_name             = "${var.domain_name}"
     zookeeper_data_dir_size = "${var.zookeeper_data_dir_size}"
     zookeeper_log_dir_size  = "${var.zookeeper_log_dir_size}"
+    storage_class           = "${var.storage_class}"
+    kafka_loadbalancers     = "${var.kafka_loadbalancers}"
   }
 }
 
-data "kubernetes_service" "lb0" {
-  metadata {
-    name      = "confluent-0-loadbalancer"
-    namespace = "${kubernetes_namespace.kafka.metadata.0.name}"
-  }
-
-  depends_on = ["helm_release.confluent"]
-}
-
-data "kubernetes_service" "lb1" {
-  metadata {
-    name      = "confluent-1-loadbalancer"
-    namespace = "${kubernetes_namespace.kafka.metadata.0.name}"
-  }
-
-  depends_on = ["helm_release.confluent"]
-}
+data "kubernetes_service" "lb" {
+  count = "${var.kafka_loadbalancers}"
 
-data "kubernetes_service" "lb2" {
   metadata {
-    name      = "confluent-2-loadbalancer"
+    name      = "confluent-${count.index}-loadbalancer"
     namespace = "${kubernetes_namespace.kafka.metadata.0.name}"
   }
 
   depends_on = ["helm_release.confluent"]
 }
 
 locals {
-  confluent_lb0_ip = "${lookup(data.kubernetes_service.lb0.load_balancer_ingress[0], "ip")}"
-  confluent_lb1_ip = "${lookup(data.kubernetes_service.lb1.load_balancer_ingress[0], "ip")}"
-  confluent_lb2_ip = "${lookup(data.kubernetes_service.lb2.load_balancer_ingress[0], "ip")}"
+  confluent_lb_ips = ["${data.kubernetes_service.lb.0.load_balancer_ingress.0.ip}"]
```

Comment:
Was this tested with a `count` of > than 1?  I believe a `*` should be required.

Reply:
So I tried `confluent_lb_ips = ["${data.kubernetes_service.lb.*.load_balancer_ingress.0.ip}"] ` as suggested [here](https://www.terraform.io/docs/configuration-0-11/interpolation.html#attributes-of-a-data-source) but it didn't work 
```
Error: Error running plan: 1 error(s) occurred:

* module.efd.local.confluent_lb_ips: local.confluent_lb_ips: Resource 'data.kubernetes_service.lb' does not have attribute 'load_balancer_ingress.0.ip' for variable 'data.kubernetes_service.lb.*.load_balancer_ingress.0.ip'
```

using  ``confluent_lb_ips = ["${data.kubernetes_service.lb.0.load_balancer_ingress.0.ip}"]`` seems to work.

Output of `terraform plan`:

```Terraform will perform the following actions:

 <= module.efd.data.kubernetes_service.lb[0]
      id:                          <computed>
      load_balancer_ingress.#:     <computed>
      metadata.#:                  "1"
      metadata.0.generation:       <computed>
      metadata.0.name:             "confluent-0-loadbalancer"
      metadata.0.namespace:        "kafka"
      metadata.0.resource_version: <computed>
      metadata.0.self_link:        <computed>
      metadata.0.uid:              <computed>
      spec.#:                      <computed>

 <= module.efd.data.kubernetes_service.lb[1]
      id:                          <computed>
      load_balancer_ingress.#:     <computed>
      metadata.#:                  "1"
      metadata.0.generation:       <computed>
      metadata.0.name:             "confluent-1-loadbalancer"
      metadata.0.namespace:        "kafka"
      metadata.0.resource_version: <computed>
      metadata.0.self_link:        <computed>
      metadata.0.uid:              <computed>
      spec.#:                      <computed>

 <= module.efd.data.kubernetes_service.nginx_ingress
      id:                          <computed>
      load_balancer_ingress.#:     <computed>
      metadata.#:                  "1"
      metadata.0.generation:       <computed>
      metadata.0.name:             "nginx-ingress-controller"
      metadata.0.namespace:        "nginx-ingress"
      metadata.0.resource_version: <computed>
      metadata.0.self_link:        <computed>
      metadata.0.uid:              <computed>
      spec.#:                      <computed>

  ~ module.efd.aws_route53_record.grafana
      records.#:                   "" => <computed>

  ~ module.efd.aws_route53_record.influxdb
      records.#:                   "" => <computed>

  ~ module.efd.aws_route53_record.kafka_lb[0]
      records.#:                   "" => <computed>

  + module.efd.aws_route53_record.kafka_lb[1]
      id:                          <computed>
      allow_overwrite:             "true"
      fqdn:                        <computed>
      name:                        "test-efd1.lsst.codes"
      records.#:                   <computed>
      ttl:                         "60"
      type:                        "A"
      zone_id:                     "Z3TH0HRSNU67AM"
```

however, I did not test this against `gke` yet.